pulp_ansible/app/galaxy/v1/views.py

```diff
@@ -0,0 +1,136 @@
+import datetime
+
+from rest_framework import viewsets
+
+from django.shortcuts import get_object_or_404
+from rest_framework.response import Response
+
+from pulpcore.plugin.models import Task
+from pulpcore.plugin.tasking import dispatch
+
+from pulp_ansible.app.models import AnsibleDistribution, AnsibleRepository, Role
+from pulp_ansible.app.galaxy.serializers import GalaxyRoleSerializer
+from pulp_ansible.app.galaxy.v1.constants import LEGACY_REPOSITORY_NAME
+from pulp_ansible.app.galaxy.v1.tasks import legacy_role_import
+
+
+STATE_MAP = {"COMPLETED": "SUCCESS"}
+
+
+TYPE_MAP = {"RUNNING": "INFO", "WAITING": "INFO", "COMPLETED": "SUCCESS"}
+
+
+class LegacyImportView(viewsets.ModelViewSet):
+    """
+    APIView for Roles.
+    """
+
+    model = Role
+    serializer_class = GalaxyRoleSerializer
+    authentication_classes = []
+    permission_classes = []
+
+    def get_queryset(self):
+        """
+        Get the list of items for this view.
+        """
+        # distro = get_object_or_404(AnsibleDistribution, base_path=self.kwargs["path"])
+        distro = get_object_or_404(AnsibleDistribution, base_path="legacy")
```

Comment:
This is likely just something to come later, but I imagined `base_path` would come from the LEGACY_REPOSITORY_NAME somehow.

Reply:
Yeah, I missed this one.